src/onnx/parse_matmul.cpp

```diff
@@ -38,6 +38,100 @@ struct parse_matmul : op_parser<parse_matmul>
+            auto output_lens = compute_broadcasted_lens(l0_broadcasted_lens, l1_broadcasted_lens);
+            l0_broadcasted_lens = output_lens;
+            l0_broadcasted_lens.insert(l0_broadcasted_lens.end(), l0_it, s0_lens.end());
+            l1_broadcasted_lens = output_lens;
+            l1_broadcasted_lens.insert(l1_broadcasted_lens.end(), l1_it, s1_lens.end());
+            if(s0_lens != l0_broadcasted_lens)
+            {
+                ba0 = info.add_instruction(
+                    make_op("multibroadcast", {{"out_lens", l0_broadcasted_lens}}), a0);
+            }
+            if(s1_lens != l1_broadcasted_lens)
+            {
+                ba1 = info.add_instruction(
+                    make_op("multibroadcast", {{"out_lens", l1_broadcasted_lens}}), a1);
+            }
+        }
+    }
+
+    static void
+    add_int8_shift(const onnx_parser::node_info& info, instruction_ref& ba0, instruction_ref& ba1)
+    {
+        const auto ba0_type = ba0->get_shape().type();
+        const auto ba1_type = ba1->get_shape().type();
+
+        // determine if we need to add a shift to ensure items get shifted to the same int8
+        // type if we see uint8/int8 mismatches
+        if(ba0_type == ba1_type)
+            return;
+
+        instruction_ref unshifted_input;
+        instruction_ref* shifted_output = nullptr;
+        if(ba0_type == migraphx::shape::int8_type)
+        {
+            unshifted_input = ba1;
+            shifted_output  = &ba1;
+        }
+        if(ba1_type == migraphx::shape::int8_type)
+        {
+            unshifted_input = ba0;
+            shifted_output  = &ba0;
+        }
+
+        // Convert to fp16 prior to a shift to ensure we preserve accuracy here then
+        // convert back to int8
+        auto int8_shift = info.add_literal(
+            migraphx::literal{migraphx::shape{migraphx::shape::int16_type}, {-128}});
+
+        auto unshifted_input_int16 = info.add_instruction(
+            migraphx::make_op("convert", {{"target_type", migraphx::shape::int16_type}}),
+            unshifted_input);
+
+        auto input_shifted_int16 = info.add_common_op("add", unshifted_input_int16, int8_shift);
+
+        auto shifted_input = info.add_instruction(
+            migraphx::make_op("convert", {{"target_type", migraphx::shape::int8_type}}),
+            input_shifted_int16);
+
+        *shifted_output = shifted_input;
+    }
+
+    static void set_bias_arg(const onnx_parser::node_info& info,
+                             const std::vector<instruction_ref>& args,
+                             const int index,
+                             const instruction_ref& input,
+                             instruction_ref& bias_arg)
+    {
+        if(args.size() > index)
+        {
+            bias_arg = args[index];
+            if(bias_arg->get_shape().type() != input->get_shape().type())
+            {
+                MIGRAPHX_THROW("PARSE_QUANT_DOT: zero point must be the same type as data");
```

Comment:
if not too much of work add a test coverage for this line.

Reply:
Good point. Adding additional coverage then

Reply:
Coverage added